python/GafferUI/VectorDataWidget.py

```diff
@@ -418,32 +423,41 @@ def editSignal( self ) :
 
 		return self.__editSignal
 
+	## A signal emitted when the user right clicks selected cells.
+	# Slots should accept ( VectorDataWidget, IECore.MenuDefinition ) arguments,
+	# modify the given menu and return it.
+	def dataMenuSignal( self ) :
+
+		return self.__dataMenuSignal
+
 	## Returns a definition for the popup menu - this is called each time
 	# the menu is displayed to allow menus to be built dynamically. May be
 	# overridden in derived classes to modify the menu.
 	## \todo We should remove this as part of implementing #217, and just
 	# let everything hook onto contextMenuSignal() instead.
-	def _contextMenuDefinition( self, selectedRows ) :
+	def _contextMenuDefinition( self, selectedRows, menuDefinition = None ) :
 
-		m = IECore.MenuDefinition()
+		if menuDefinition is None :
+			menuDefinition = IECore.MenuDefinition()
 
-		m.append( "/Select All", { "command" : self.__selectAll } )
-		m.append( "/Clear Selection", { "command" : self.__clearSelection } )
+		menuDefinition.append( "/Select All", { "command" : self.__selectAll } )
+		menuDefinition.append( "/Clear Selection", { "command" : self.__clearSelection } )
 
 		if self.getEditable() and self.getSizeEditable() :
 
-			m.append( "/divider", { "divider" : True } )
-			m.append(
+			menuDefinition.append( "/divider", { "divider" : True } )
+			menuDefinition.append(
 				"/Delete Selected Rows",
 				{
 					"command" : functools.partial( Gaffer.WeakMethod( self.__removeRows ), selectedRows ),
 					"shortCut" : "Backspace, Delete"
 				}
 			)
 
-		return m
+		return menuDefinition
 
-	## This function is used by the ui to create new data to append. It may be overridden
+	## \deprecated Use dataMenuSignal() instead.
```

Comment:
I don't see how this is related to `_createRows()`, which is called from the <kbd>+</kbd> button rather than the menu. I think we want to deprecated `_contextMenuDefinition()` instead?

Reply:
Fixed and squashed into 71a0621e4ce0494c03be56e1315fa5a62a9ff1f2